test/helpers/utils.go

```diff
@@ -120,6 +122,73 @@ func WaitForPoolComplete(t *testing.T, cs *framework.ClientSet, pool, target str
 	return nil
 }
 
+// WaitForPoolComplete polls a pool until it has completed any update
+func WaitForPoolCompleteAny(t *testing.T, cs *framework.ClientSet, pool string) error {
```

Comment:
Something to be mindful about is if you run this test in an SNO context, this function will prematurely error because the control plane may go offline.

To work around that, you'd need to do something like: https://github.com/openshift/machine-config-operator/blob/master/test/e2e-single-node/sno_mcd_test.go#L355-L374 or: https://github.com/openshift/machine-config-operator/pull/2795/files#diff-b240aacd848b27ec33331f84d7e3fc0a4bc20045f45f2cd536727396a756e97bR195-R223.

The short version is:
1. Ignore the error you get from the `cs.MachineConfigPools().Get()`.
2. Run the clock out.
3. If you're still getting an error from `cs.MachineConfigPools().Get()`, then fail.

Reply:
Excellent. Thanks. I'll rearrange the test after your helpers merge. I did run the test against single node, so I know it works (aside from handling the control plane conditions you're describing).

Reply:
Agree with Zack that for SNO, cluster can be itself unreachable because everything is running on the same node and when node reboots, it will be unreachable for a while. However, for this particular test case I won't be worried much because applying kube cert shouldn't cause drain or reboot. Ideally, cluster availability shouldn't be impacted and if it does happen, something is wrong.